src/components/Modal/types.ts

```diff
@@ -81,7 +96,23 @@ export type ModalProps = {
   horizontalPadding?: number
 
   /**
-   * Horizontal padding between outside of modal and inner content
+   * @deprecated in favor of composability - use ModalContent sub-component instead
    */
   contentHorizontalPadding?: number
 }
+
+export type ModalContentProps = React.HTMLAttributes<HTMLDivElement>
+
+export type ModalHeaderProps = React.HTMLAttributes<HTMLDivElement> & {
+  showDismissButton?: boolean
+  onClose?: () => void
+  children: React.ReactNode
+}
+
+export type ModalTitleProps = React.HTMLAttributes<HTMLDivElement> & {
+  subtitleClassName?: string
+  icon?: React.ReactNode
```

Comment:
personally, i like importing ReactNode and HTMLAttributes as named imports from 'react'